src/EntityComponentManager.cc

```diff
@@ -94,6 +94,12 @@ class ignition::gazebo::EntityComponentManagerPrivate
 
   /// \brief Keep track of entities already used to ensure uniqueness.
   public: uint64_t entityCount{0};
+
+  /// \brief Map of removed components. The key is the entity to
+  /// which belongs the component, and the value is the component being
+  /// removed. This is set when RemoveComponent() is called, so we assume
+  /// only one removed component per entity.
+  std::map<Entity, ComponentKey> removedComponents;
```

Comment:
I don't really know how the sync works, but I supposed that it would occur immediately after `RemoveComponent` is called. So if you want to remove two different components of the same entity you would have to call `RemoveComponent` twice. My question is, in this case sync would take place after the first call or after the second?

Reply:
No, I believe sync happens once every iteration when the `SceneBroadcaster` publishes the state message in its `PostUpdate` https://github.com/ignitionrobotics/ign-gazebo/blob/139020c8ec0af219efd848b84b4954c2c65936e9/src/systems/scene_broadcaster/SceneBroadcaster.cc#L302 So, calling `RemoveComponent` doesn't cause a sync.

Reply:
7992720